File: solutions/python/structure/fixed/StaticPriorityQueueADT.py
class StaticPriorityQueueADT:
    def __init__(self, capacidad: int):
        self.capacidad = capacidad
        self.datos = [None] * capacidad  # Cada elemento será una tupla (valor, prioridad)
        self.size = 0

    def get_element(self) -> int:
        if self.is_empty():
            raise Exception("No se puede obtener el primero de una cola vacía")
        return self.datos[0][0]

    def get_priority(self) -> int:
        if self.is_empty():
            raise Exception("No se puede obtener la prioridad del primero de una cola vacía")
        return self.datos[0][1]

    def remove(self):
        if self.is_empty():
            raise Exception("No se puede remover el primero de una cola vacía")
        for i in range(1, self.size):
            self.datos[i - 1] = self.datos[i]
        self.datos[self.size - 1] = None
        self.size -= 1

    def add(self, value: int, priority: int):
        if self.size >= self.capacidad:
            raise Exception("Cola llena")

        i = self.size - 1
        # Insertamos ordenado por prioridad ascendente
        while i >= 0 and self.datos[i][1] > priority:
            self.datos[i + 1] = self.datos[i]
            i -= 1
        self.datos[i + 1] = (value, priority)
        self.size += 1

    def is_empty(self) -> bool:
        return self.size == 0

    def mostrar(self):
        for i in range(self.size):
            print(f"[{self.datos[i][0]}, {self.datos[i][1]}]", end=" -> ")
        print("None")

File: solutions/javascript/src/structure/fixed/StaticPriorityQueueADT.js
class StaticPriorityQueueADT {
  constructor(capacity) {
    this.capacity = capacity;
    this.values = new Array(capacity);
    this.priorities = new Array(capacity);
    this.size = 0;
  }

  // Agrega un elemento según su prioridad
  add(value, priority) {
    if (this.size >= this.capacity) {
      throw new Error("La cola ha alcanzado su capacidad máxima");
    }

    let i = this.size - 1;
    // Mover elementos con menor prioridad a la derecha
    while (i >= 0 && this.priorities[i] > priority) {
      this.values[i + 1] = this.values[i];
      this.priorities[i + 1] = this.priorities[i];
      i--;
    }

    // Insertar el nuevo valor en la posición correcta
    this.values[i + 1] = value;
    this.priorities[i + 1] = priority;
    this.size++;
  }

  // Devuelve el valor del primer elemento
  getElement() {
    if (this.isEmpty()) {
      throw new Error("La cola está vacía");
    }
    return this.values[0];
  }

  // Devuelve la prioridad del primer elemento
  getPriority() {
    if (this.isEmpty()) {
      throw new Error("La cola está vacía");
    }
    return this.priorities[0];
  }

  // Elimina el primer elemento
  remove() {
    if (this.isEmpty()) {
      throw new Error("La cola está vacía");
    }
    for (let i = 1; i < this.size; i++) {
      this.values[i - 1] = this.values[i];
      this.priorities[i - 1] = this.priorities[i];
    }
    this.size--;
  }

  // Devuelve true si no hay elementos
  isEmpty() {
    return this.size === 0;
  }
}

export default StaticPriorityQueueADT;